[Issue]: Incompatible with CMake 4

### Problem Description

https://github.com/ROCm/hipfort/blob/d50f85a777cad4811350925c978e0d9ac5565c09/bin/CMakeLists.txt#L2

hipfort sets cmake_minimum_required to too old a value

```
       >
       > 
       > CMake Error at bin/CMakeLists.txt:2 (cmake_minimum_required):
       >   Compatibility with CMake < 3.5 has been removed from CMake.
       >
       >   Update the VERSION argument <min> value.  Or, use the <min>...<max> syntax
       >   to tell CMake that the project requires at least <min> but has been updated
       >   to work with policies introduced by <max> or earlier.
       >
       >   Or, add -DCMAKE_POLICY_VERSION_MINIMUM=3.5 to try configuring anyway.
       >
       > 
       > -- Configuring incomplete, errors occurred!
```

This can be fixed by updating the value. 3.5 is the minimum allowed for 4.x compat.

I've tested all the way up to `CMAKE_POLICY_VERSION_MINIMUM=4.0` so it should be safe to replace it with `cmake_minimum_required(VERSION 3.0..4.0 FATAL_ERROR)` for futureproofing.

### Operating System

NixOS

### CPU

EPYC 7713

### ROCm Version

ROCm 6.4.3

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,4 +1,4 @@
-CMAKE_MINIMUM_REQUIRED(VERSION 3.18...3.31)
+cmake_minimum_required(VERSION 3.18..4.0 FATAL_ERROR)
 
 #Setting Default value for libdir
 set(CMAKE_INSTALL_LIBDIR "lib" CACHE STRING "Library install directory")

--- a/bin/CMakeLists.txt
+++ b/bin/CMakeLists.txt
@@ -1,5 +1,4 @@
-
-cmake_minimum_required(VERSION 2.8.12 FATAL_ERROR)
+cmake_minimum_required(VERSION 3.18..4.0 FATAL_ERROR)
 if ("${CMAKE_SOURCE_DIR}" STREQUAL "${CMAKE_CURRENT_SOURCE_DIR}")
   project(hipfort-tools)
 endif()

--- a/test/CMakeLists.txt
+++ b/test/CMakeLists.txt
@@ -5,7 +5,7 @@
 # Each test is contained in its own subdirectory of this test directory
 # The Makefile must create a binary with the same name as the directory name.
 
-cmake_minimum_required(VERSION 3.5)
+cmake_minimum_required(VERSION 3.18..4.0 FATAL_ERROR)
 if ("${CMAKE_SOURCE_DIR}" STREQUAL "${CMAKE_CURRENT_SOURCE_DIR}")
   project(test-hipfort)
 endif()

